diff --git a/node_modules/solidity-coverage/lib/app.js b/node_modules/solidity-coverage/lib/app.js
index 6e08eda..30ee172 100644
--- a/node_modules/solidity-coverage/lib/app.js
+++ b/node_modules/solidity-coverage/lib/app.js
@@ -121,7 +121,7 @@ class App {
         // Put the coverage network in the existing config
         if (!truffleConfig.networks) truffleConfig.networks = {};
         truffleConfig.networks.coverage = coverageNetwork;
-        const configString = `module.exports = ${JSON.stringify(truffleConfig)}`;
+        const configString = `require("ts-node/register"); module.exports = ${JSON.stringify(truffleConfig)}`;
         fs.writeFileSync(`${this.coverageDir}/truffle-config.js`, configString);
       }
 
