diff --git a/node_modules/typechain/dist/parser/typeParser.js b/node_modules/typechain/dist/parser/typeParser.js
index 93d5265..f5a1959 100644
--- a/node_modules/typechain/dist/parser/typeParser.js
+++ b/node_modules/typechain/dist/parser/typeParser.js
@@ -89,7 +89,7 @@ function parseEvmType(rawType, components) {
         case "byte":
             return new BytesType(1);
         case "bytes":
-            return new ArrayType(new BytesType(1));
+            return new BytesType(1);
         case "tuple":
             if (!components)
                 throw new Error("Tuple specified without components!");
diff --git a/node_modules/typechain/dist/targets/truffle/generation.js b/node_modules/typechain/dist/targets/truffle/generation.js
index 893422e..4255ed6 100644
--- a/node_modules/typechain/dist/targets/truffle/generation.js
+++ b/node_modules/typechain/dist/targets/truffle/generation.js
@@ -4,7 +4,8 @@ const typeParser_1 = require("../../parser/typeParser");
 function codegen(contracts) {
     const template = `
 /// <reference types="truffle-typings" />
-import { BigNumber } from "bignumber.js";
+import BN from "bn.js";
+type BigNumber = string | number | BN;
 
 ${contracts.map(generateContractInterface).join("\n")}
 
